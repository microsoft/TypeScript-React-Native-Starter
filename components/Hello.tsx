import React from "react"
import { Button, StyleSheet, Text, View } from "react-native"

export interface Props {
  name: string
  enthusiasmLevel?: number
}

interface State {
  enthusiasmLevel: number
}

export class Hello extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)

    if ((props.enthusiasmLevel || 0) <= 0) {
      throw new Error("You could be a little more enthusiastic. :D")
    }

    this.state = {
      enthusiasmLevel: props.enthusiasmLevel || 1
    }
  }

  onIncrement = () => this.setState({ enthusiasmLevel: this.state.enthusiasmLevel + 1 });
  onDecrement = () => this.setState({ enthusiasmLevel: this.state.enthusiasmLevel - 1 });
  getExclamationMarks = (numChars: number) => Array(numChars + 1).join("!")

  render() {
    return (
      <View style={styles.root}>
        <Text style={styles.greeting}>
          Hello {this.props.name + this.getExclamationMarks(this.state.enthusiasmLevel)}
        </Text>
        
        <View style={styles.buttons}>
          <View style={styles.button}>
            <Button
              title="-"
              onPress={this.onDecrement}
              accessibilityLabel="decrement"
              color="red"
            />
          </View>

          <View style={styles.button}>
            <Button
              title="+"
              onPress={this.onIncrement}
              accessibilityLabel="increment"
              color="blue"
            />
          </View>
        </View>
      </View>
    )
  }
}

// styles

const styles = StyleSheet.create({
  root: {
    alignItems: "center",
    alignSelf: "center"
  },
  buttons: {
    flexDirection: "row",
    minHeight: 70,
    alignItems: "stretch",
    alignSelf: "center",
    borderWidth: 5
  },
  button: {
    flex: 1,
    paddingVertical: 0
  },
  greeting: {
    color: "#999",
    fontWeight: "bold"
  }
})
