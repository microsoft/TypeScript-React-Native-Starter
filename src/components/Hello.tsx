import React from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';

interface Props {
  name: string;
}

interface State {
  date: Date;
}

export class Hello extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { date: new Date() };
  }
  private updateDate = () => {
    this.setState({
      date: new Date()
    })
  };

  render() {
    const { name, } = this.props;

    return (
      <View style={styles.root}>
        <Text style={styles.greeting}>
          Hello {name + '!'}
        </Text>
        <Text style={styles.greeting}>
          Current Time: {this.state.date.toLocaleString()}
        </Text>
        <View style={styles.buttons}>
          <View style={styles.button}>
            <Button title="Update Time" onPress={this.updateDate} accessibilityLabel="decrement" color='red' />
          </View>
        </View>
      </View>
    );
  }
}

// styles

const styles = StyleSheet.create({
  root: {
    alignItems: "center",
    alignSelf: "center"
  },
  buttons: {
    flexDirection: "row",
    minHeight: 70,
    alignItems: "stretch",
    alignSelf: "center",
    borderWidth: 5,
  },
  button: {
    flex: 1,
    paddingVertical: 0,
  },
  greeting: {
    color: "#999",
    fontWeight: "bold"
  }
});
