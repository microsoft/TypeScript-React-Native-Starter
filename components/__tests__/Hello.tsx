import React from 'react'
import renderer from 'react-test-renderer'

import { Hello } from '../Hello'

it('renders correctly with defaults', () => {
    const button = renderer.create(<Hello name="World" enthusiasmLevel={1} />).toJSON()
    expect(button).toMatchSnapshot()
})
