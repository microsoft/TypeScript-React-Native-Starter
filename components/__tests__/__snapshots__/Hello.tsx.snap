// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#999",
        "fontWeight": "bold",
      }
    }
  >
    Hello 
    World!
  </Text>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "alignSelf": "center",
        "borderWidth": 5,
        "flexDirection": "row",
        "minHeight": 70,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 0,
        }
      }
    >
      <View
        accessibilityLabel="decrement"
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "red",
                },
              ]
            }
          >
            -
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 0,
        }
      }
    >
      <View
        accessibilityLabel="increment"
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "blue",
                },
              ]
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
